cloudpickle/cloudpickle_fast.py

```diff
@@ -0,0 +1,554 @@
+"""
+New, fast version of the CloudPickler.
+
+This new CloudPickler class can now extend the fast C Pickler instead of the
+previous python implementation of the Pickler class. Because this functionality
+is only available for python versions 3.8+, a lot of backward-compatibility
+code is also removed.
+"""
+import abc
+import copyreg
+import io
+import itertools
+import logging
+import _pickle
+import pickle
+import sys
+import types
+import weakref
+
+from _pickle import Pickler
+
+from .cloudpickle import (
+    _is_dynamic, _extract_code_globals, _BUILTIN_TYPE_NAMES, DEFAULT_PROTOCOL,
+    _find_imported_submodules, _get_cell_contents, _is_global, _builtin_type,
+    Enum, _ensure_tracking, _lookup_class_or_track, _make_skeleton_class,
+    _make_skeleton_enum, _extract_class_dict, string_types, dynamic_subimport,
+    subimport
+)
+
+load, loads = _pickle.load, _pickle.loads
+
+# Shorthands similar to pickle.dump/pickle.dumps
+def dump(obj, file, protocol=None):
+    """Serialize obj as bytes streamed into file
+
+    protocol defaults to cloudpickle.DEFAULT_PROTOCOL which is an alias to
+    pickle.HIGHEST_PROTOCOL. This setting favors maximum communication speed
+    between processes running the same Python version.
+
+    Set protocol=pickle.DEFAULT_PROTOCOL instead if you need to ensure
+    compatibility with older versions of Python.
+    """
+    CloudPickler(file, protocol=protocol).dump(obj)
+
+
+def dumps(obj, protocol=None):
+    """Serialize obj as a string of bytes allocated in memory
+
+    protocol defaults to cloudpickle.DEFAULT_PROTOCOL which is an alias to
+    pickle.HIGHEST_PROTOCOL. This setting favors maximum communication speed
+    between processes running the same Python version.
+
+    Set protocol=pickle.DEFAULT_PROTOCOL instead if you need to ensure
+    compatibility with older versions of Python.
+    """
+    with io.BytesIO() as file:
+        cp = CloudPickler(file, protocol=protocol)
+        cp.dump(obj)
+        return file.getvalue()
+
+
+# COLLECTION OF OBJECTS __getnewargs__-LIKE METHODS
+# -------------------------------------------------
+
+def _class_getnewargs(obj):
+    # On PyPy, __doc__ is a readonly attribute, so we need to include it in
+    # the initial skeleton class.  This is safe because we know that the
+    # doc can't participate in a cycle with the original class.
+    type_kwargs = {'__doc__': obj.__dict__.get('__doc__', None)}
+
+    if hasattr(obj, "__slots__"):
+        type_kwargs["__slots__"] = obj.__slots__
+
+    __dict__ = obj.__dict__.get('__dict__', None)
+    if isinstance(__dict__, property):
+        type_kwargs['__dict__'] = __dict__
```

Comment:
This case is not covered by the tests:

https://codecov.io/gh/cloudpipe/cloudpickle/compare/f3c3aeae28a650f3ed0ea5f8516e2f090ce44536...3b879044a2e4d46350baf9e77a39da63355d4ae7/diff

Reply:
Yes. It may be 2.7 specific. As this seems covered in `cloudpickle.py`.

Reply:
But this specific line is in `cloudpickle/cloudpickle_fast.py ` not `cloudpickle/cloudpickle.py `

Reply:
This is not resolved yet, but this requires do dive deep into python's way to look for object's attribute.